pkg/bmc/virtualmedia_validation.go

```diff
@@ -0,0 +1,100 @@
+package bmc
+
+import (
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/stmcginnis/gofish"
+	"github.com/stmcginnis/gofish/redfish"
+)
+
+// contains tells whether the array of redfish.VirtualMediaType
+// has the element redfish.VirtualMediaType.
+func contains(array []redfish.VirtualMediaType, element redfish.VirtualMediaType) bool {
+	for _, n := range array {
+		if element == n {
+			return true
+		}
+	}
+	return false
+}
+
+// ValidateVirtualMedia returns nil when the BMC has support
+// for VirtualMedia and error when the BMC does not have support.
+func ValidateVirtualMedia(address string, user string, password string, insecure bool, httpClient *http.Client) error {
+	bmcConfig := gofish.ClientConfig{
+		Endpoint:   address,
+		Username:   user,
+		Password:   password,
+		Insecure:   insecure,
+		HTTPClient: httpClient,
+	}
+
+	client, err := gofish.Connect(bmcConfig)
+	if err != nil {
```

Comment:
I'm unsure if this should be an error. I guess since the BMO container is co-located with ironic, the validation should normally succeed, but do we want to support cases of external ironic?

Reply:
@andfasano @dtantsur  we need to also remember the function `ValidateVirtualMedia` will be used outside BMO (in the installer)

Reply:
This issue has not been resolved. I'm not sure how the installer comment is relevant here: inability to connect to the BMC cannot count as an error.

Reply:
Since the installer will use this function we need to identify when it can't access the BMC (that's why I'm rasing `AccessVirtualMediaValidationError`. If we can't connect to the BMC the validation will fail on BMO, or we should ignore the error?

Reply:
@dtantsur , (this is probably more clear, please ignore my last comment) the `inability to connect to the BMC cannot count as an error` in the installer side when we will try to validate (because not always the installer can reach the BMC network), so we will give a message that we will skip the validation because we couldn't access the BMC.

Reply:
Yeah, In the installer we will consider a warning =)